weed/topology/topology_event_handling.go

```diff
@@ -23,9 +24,12 @@ func (t *Topology) StartRefreshWritableVolumes(grpcDialOption grpc.DialOption, g
 	}()
 	go func(garbageThreshold float64) {
 		c := time.Tick(15 * time.Minute)
-		for _ = range c {
+		for range c {
 			if t.IsLeader() {
+				time.Sleep(time.Duration(120*rand.Float32()) * time.Second)
 				t.Vacuum(grpcDialOption, garbageThreshold, 0, "", preallocate)
+			} else {
+				stats.MasterReplicaPlacementMismatch.Reset()
```

Comment:
this seems an orthoganal change.

This will reset the stats every 15 minutes?

Reply:
Only the leader keeps statistics.  And at feasts you need a reset.  There's a for loop inside the labels in the map

Reply:
If only the leader keeps statistics, why the non-leaders needs to reset the stats?

Reply:
> If only the leader keeps statistics, why the non-leaders needs to reset the stats?

Since the metrics are collected from all masters, and the ReplicaPlacementMismatch metric may show an outdated value from the cache

```
SeaweedFS_master_replica_placement_mismatch{collection="files", endpoint="sw-master-exporter", id="299", instance="172.24.196.5:9090", job="prometheus-operator-job", namespace="s3", pod="fast-master-2", service="fast-master"}
1
SeaweedFS_master_replica_placement_mismatch{collection="files", endpoint="sw-master-exporter", id="299", instance="172.24.221.214:9090", job="prometheus-operator-job", namespace="s3", pod="fast-master-1", service="fast-master"}
0

SeaweedFS_master_is_leader{endpoint="sw-master-exporter", instance="172.24.164.250:9090", job="prometheus-operator-job", namespace="s3", pod="fast-master-0", service="fast-master"} 0
SeaweedFS_master_is_leader{endpoint="sw-master-exporter", instance="172.24.196.5:9090", job="prometheus-operator-job", namespace="s3", pod="fast-master-2", service="fast-master"} | 0
SeaweedFS_master_is_leader{endpoint="sw-master-exporter", instance="172.24.221.214:9090", job="prometheus-operator-job", namespace="s3", pod="fast-master-1", service="fast-master"} 1
```

Reply:
@chrislusf Did I manage to explain?